### PR TITLE
IDENTITY-5231 Fixing response subject, incorrect for oauth userinfo e…

### PR DESCRIPTION
**Pull request by dimuthud**
…ndpoint

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 1]
```diff
@@ -158,7 +168,64 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
...
+
+            String accessTokenClientId;
+            try {
+                accessTokenClientId = OAuth2Util.getClientIdForAccessToken
+                        (tokenResponse.getAuthorizationContextToken().getTokenString());
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error while obtaining service provider access token clientID ", e);
+                throw new UserInfoEndpointException("Error while obtaining service provider access token clientID", e);
+            }
+
```
Comment (isharak) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: shall we put this in to a private method, and please add more comments

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 2]
```diff
@@ -158,7 +168,64 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
...
         }
         if (!returnClaims.containsKey("sub") || StringUtils.isBlank((String) claims.get("sub"))) {
-            returnClaims.put("sub", tokenResponse.getAuthorizedUser());
+
+            String accessTokenClientId;
+            try {
+                accessTokenClientId = OAuth2Util.getClientIdForAccessToken
+                        (tokenResponse.getAuthorizationContextToken().getTokenString());
+            } catch (IdentityOAuth2Exception e) {
+                log.error("Error while obtaining service provider access token clientID ", e);
```
Comment (isharak) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: don't log an throw either log or throw

Comment (dimuthud) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Logs removed.

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java (hunk 1 above)
Comment (dimuthud) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Added comments.

**Comment by JKAUSHALYA** [comment]
Closing until conflicts are resolved.

**Pull request closed without merging**